src/OrbitGl/include/OrbitGl/SelectionData.h

```diff
@@ -0,0 +1,58 @@
+
+// Copyright (c) 2023 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef CLIENT_DATA_SELECTION_DATA_H_
+#define CLIENT_DATA_SELECTION_DATA_H_
+
+#include "ClientData/CaptureData.h"
+#include "ClientData/ModuleManager.h"
+#include "ClientData/PostProcessedSamplingData.h"
+#include "OrbitGl/CallTreeView.h"
+
+// This is meant to hold the data needed to update the data tabs to a specific selection.
+class SelectionData {
+ public:
+  // Delete move/copy operators because callstack_data_pointer_ is pointing to a variable inside the
+  // class.
+  SelectionData(const SelectionData& other) = delete;
+  SelectionData& operator=(const SelectionData& other) = delete;
+  SelectionData(SelectionData&& other) = delete;
+  SelectionData& operator=(SelectionData&& other) = delete;
+
+  SelectionData(const orbit_client_data::ModuleManager& module_manager,
+                const orbit_client_data::CaptureData& capture_data,
+                orbit_client_data::PostProcessedSamplingData post_processed_sampling_data,
+                const orbit_client_data::CallstackData* callstack_data);
+
+  SelectionData(const orbit_client_data::ModuleManager& module_manager,
+                const orbit_client_data::CaptureData& capture_data,
+                absl::Span<const orbit_client_data::CallstackEvent> callstack_events);
+
+  std::shared_ptr<const CallTreeView> GetTopDownView() const { return top_down_view_; }
+
+  std::shared_ptr<const CallTreeView> GetBottomUpView() const { return bottom_up_view_; }
+
+  const orbit_client_data::PostProcessedSamplingData& GetPostProcessedSamplingData() const {
+    return post_processed_sampling_data_;
+  }
+
+  const orbit_client_data::CallstackData& GetCallstackData() const {
+    ORBIT_CHECK(callstack_data_pointer_);
+    return *callstack_data_pointer_;
+  }
+
+ private:
+  std::shared_ptr<CallTreeView> top_down_view_;
```

Comment:
Can we somehow make this a unique pointer?

Reply:
We could make it a unique ptr and give const pointers in the getters, but that makes it really easy to have dangling pointers and have crashes like the one with callstack data.  This class should make it easier to keep track of the selections and make sure the views are left with any dangling pointers. the way the code is written now, it would be fine because it sets the views before overwriting the pointers in app, but it's fragile if it gets changed whereas it is more robust with it being a shared ptr and the caller doesn't have to worry about the order in which you overwrite the selection data and update the views.

Reply:
Yes but I think that destroys the point; One purpose is to take care of life time issues; while right now it would still work to make it a unique ptr and then pass const pointers to the views, SelectionData is getting overwritten a lot and it matters which order you create a new SelectionData, hand it over to the views, and overwrite the old SelectionData. Keeping it shared prevents crashes similar to the CallstackData crash we were working on for these views, although I'm still thinking about what to do about PostProcessedSampilngData/CalsltackData.  There is still a clear owner as SelectionData holds the only non-const pointer to them.